toucan_connectors/one_drive/one_drive_connector.py

```diff
@@ -174,12 +192,25 @@ def _run_fetch(self, url):
     def _retrieve_data(self, data_source: OneDriveDataSource) -> pd.DataFrame:
         logging.getLogger(__name__).debug('_retrieve_data')
 
-        df_sheet_all = pd.DataFrame()
+        if data_source.sheet and data_source.table:
+            raise ValueError('You cannot specifiy both sheets and tables')
 
-        sheet_list = data_source.sheet.replace(', ', ',').split(',')
+        if data_source.range and data_source.table:
+            raise ValueError('You cannot specify a range for tables (tables is a kind of range)')
 
-        for sheet in sheet_list:
-            url = self._format_url(data_source, sheet)
+        if not data_source.sheet and not data_source.table:
+            raise ValueError('You must specify at least a sheet or a table')
+
+        df_all = pd.DataFrame()
+
+        workbook_elements_list = []
+        if data_source.sheet:
+            workbook_elements_list = data_source.sheet.replace(', ', ',').split(',')
+        else:
+            workbook_elements_list = data_source.table.replace(', ', ',').split(',')
```

Comment:
What i suggest to you :
```suggestion
        workbook_elements_list = []
        workbook_key_column = ''
        if data_source.sheet:
            workbook_elements_list = data_source.sheet
            workbook_key_column = '__sheetname__'
        else:
            workbook_elements_list = data_source.table
            workbook_key_column = '__tablename__'

        workbook_elements_list = workbook_elements_list.replace(', ', ',').split(',')
```